### PR TITLE
Bug 1515078 - Add opt out UX in about:preferences for Pocket Newtab

### PR DESCRIPTION
**Pull request by piatra**
In order to test this 
* set `browser.newtabpage.activity-stream.discoverystream.config` to have `enable` set to true
* navigate to `about:preferences#home`

### CODE REVIEW

File: lib/AboutPreferences.jsm [hunk 1]
```diff
@@ -257,6 +260,36 @@ this.AboutPreferences = class AboutPreferences {
       });
     });
 
+    try {
+      const discoveryStreamPref = JSON.parse(ASPreferences.values["discoverystream.config"]);
+      if (discoveryStreamPref.enabled) {
```
Comment (Mardak) [inline lib/AboutPreferences.jsm]: Seems like we should probably avoid parsing the pref here. @ScottDowne would it be safe to check `state.DiscoveryStream.layout` to be non-0 length? Or even DiscoveryStream.config.enabled would avoid re-parsing json. ??

File: lib/AboutPreferences.jsm [hunk 2]
```diff
@@ -257,6 +260,36 @@ this.AboutPreferences = class AboutPreferences {
...
+        // If Discovery Stream is enabled hide Home Content options
+        contentsGroup.style.visibility = "hidden";
+
+        const discoveryGroup = homeGroup.insertAdjacentElement("afterend", homeGroup.cloneNode());
+        discoveryGroup.id = "discoveryContentsGroup";
+        discoveryGroup.setAttribute("data-subcategory", "contents");
```
Comment (Mardak) [inline lib/AboutPreferences.jsm]: Probably make the subcategory "discovery". As this is now, it prevents `about:preferences#home-contents` from working when it's not enabled (as it tries to highlight a hidden section).

File: lib/AboutPreferences.jsm (hunk 1 above)
Comment (ScottDowne) [reply lib/AboutPreferences.jsm]: `DiscoveryStream.config.enabled` would probably be ideal. I dunno if checking the length might be weird.

**Comment by piatra** [comment]
@Mardak this will require a follow up m-c patch to handle the Restore Defaults button. We still want to restore about:home/about:newtab changes but not other preferences (for Sections) that might have changed. We want the button to be hidden if only the hidden preferences have changes.

This is what I had in mind https://gist.github.com/piatra/d290a031c7aae836dfe9b1e532f79721
What do you think?

**Pull request merged**